Cryo attocube wh

andresrs9 opened the pull request:
Hello! 
This branch contains a script that moves the piezos and other one that reads the temperature of the cryo.

DavidOttoTiede committed:
Add 2D data

DavidOttoTiede committed:
Enhanced DataHandling

DavidOttoTiede committed:
Add requirements

DavidOttoTiede committed:
Add reference cryo interface

DavidOttoTiede committed:
Add reference powermeter

DavidOttoTiede committed:
Add reference powermeter

DavidOTiede committed:
Added Thorlabs PM driver. Still needs to be commented and cleaned up

DavidOTiede committed:
Added Thorlabs CCS200 working examples.

DavidOttoTiede committed:
Add 1D SpectrometerPlot and DataHandling

DavidOTiede committed:
Added Thorlabs CCS200 driver. Still with bugs.

DavidOttoTiede committed:
Add Pixis driver and ThorlabsPM100DDemo

DavidOttoTiede committed:
Add Spectrograph to Pixis interface

DavidOttoTiede committed:
Add Spectrograph to Pixis interface

DavidOTiede committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main.py

DavidOttoTiede committed:
Add Pixis test

DavidOTiede committed:
Merge remote-tracking branch 'origin/dev' into dev

DavidOTiede committed:
Make Pixis and SP2150 work on BF PC

DavidOTiede committed:
Make Pixis and SP2150 work on BF PC

DavidOttoTiede committed:
Add calibration notebook

DavidOttoTiede committed:
Merge remote-tracking branch 'origin/dev' into dev

DavidOTiede committed:
Add calibration to Pixis

DavidOttoTiede committed:
Debug data saving

DavidOttoTiede committed:
Debug data saving

DavidOttoTiede committed:
Debug data saving

DavidOTiede committed:
Improve Data saving

DavidOTiede committed:
Add T dep measurement

DavidOTiede committed:
Add T dep measurement

DavidOTiede committed:
Debug T dependence

DavidOTiede committed:
Improve DataHandling

DavidOTiede committed:
Update location temp file

DavidOTiede committed:
Add documentation to calibration.

DavidOTiede committed:
Updated Calibration

DavidOTiede committed:
Updated DataHandling. Works with 1D and 2D data

DavidOTiede committed:
Updated DataHandling. Works with 1D and 2D data

DavidOTiede committed:
Added requirements and update DataHandling

DavidOTiede committed:
Added .bat file to run Silvabot with one click

DavidOTiede committed:
Implement CCS200 globally

DavidOTiede committed:
initial commits of this branch

DavidOTiede committed:
Add hardware class for Thorlabs CCS200 spectrometer.

DavidOTiede committed:
Remove unnecessary prints

DavidOTiede committed:
Update ThorlabsPM100D to allow to select different devices. Still needs to be cleaned up.

DavidOTiede committed:
Correct bug in averaging mode of Pixis

Commit:
Added piezos driver and temperature lecture

Commit:
Added piezos driver and temperature lecture

DavidOTiede commented:
This code is part of Silvabot. It should be merged into Silvabot:dev, and not into colbert. @andresrs9, can you fix that?

DavidOTiede commented on src/drivers/Piezos.py:
```diff
@@ -0,0 +1,190 @@
+#This code is a Demo that controles attocube piezos. This routine 
```
It is not a demo, but the actual driver.

DavidOTiede committed:
Refactor comments and improve port handling

Updated comments for clarity and corrected port variable usage.

DavidOTiede reviewed (changes requested):
There is a general communication issue between the different drivers: All changes of parameters should be handled through the "set parameter" function and not through hard coded commands in the main.py file. "set parameter" needs to call the routines that make the piezos move if the position is changed.  Please remove piezo-related code from the main script and place it in the driver class, such that main.py does not get overcrowded and we make sure that the code stay modular. We can discuss in person how to adapt the code.

DavidOTiede committed:
Add two ports for Powermeters.

DavidOTiede committed:
bug fix in CCS200. Add file in .gitignore

DavidOTiede committed:
Add arduino code. Homing option still needs to be established

DavidOTiede committed:
Bug fix in CCS200

DavidOTiede committed:
Added a few comments to simplify code. Andres will complete.

Commit:
Minor changes in Lakeshore driver

Commit:
fixed merging conflict

Commit:
commited before remove autotune again

Commit:
commited before remove autotune again again

Commit:
commited before remove autotune again version 2

Commit:
commited after remove the heater autotune

DavidOTiede committed:
Adapt PM100D

DavidOTiede committed:
Clean up PM and CCS200 drivers before merging into dev.

DavidOTiede committed:
Merge pull request #2 from SilvaScience/dev_spec_arduino

This merge is accepted without further review, as David is currently the only active user of this.

DavidOTiede committed:
Rename project from Colberto to Silvabot

Updated project name and related documentation.

DavidOTiede committed:
Merge branch 'dev' into cryo_attocube_wh

DavidOTiede committed:
Update Piezos driver with improved documentation

Refactor Piezos driver code to improve documentation and clarify parameter handling. Stepsize still needs to be verified and currently no device worker is in place.

DavidOTiede committed:
Update setpoint temperature and clean up code

Updated the setpoint temperature from 270 to 300 K and cleaned up the code by removing commented lines and unnecessary whitespace.

DavidOTiede committed:
Refactor MainInterface initialization and device setup

Refactor MainInterface class to streamline initialization and device handling.

DavidOTiede committed:
Fix comment typo in update_read_parameter method

DavidOTiede committed:
Clean up Lakeshore and piezo drivers.

DavidOTiede committed:
Final commit before merge

fthouin commented:
@DavidOTiede can I close this pull request? Is this relevant to Colberto?

DavidOTiede commented:
Yes, this pull request can be closed, it was not associated to the right project
